akka-typed/src/main/scala/akka/typed/Props.scala

```diff
@@ -164,17 +164,54 @@ private[akka] case object EmptyProps extends Props {
 }
 
 /**
- * Subclasses of this type describe which thread pool shall be used to run
- * the actor to which this configuration is applied.
+ * Not intended for user extension.
+ */
+@DoNotInherit
+abstract class DispatcherSelector extends Props
+
+/**
+ * Factories for [[DispatcherSelector]]s which describe which thread pool shall be used to run
+ * the actor to which this configuration is applied. Se the individual factory methods for details
+ * on the options.
  *
  * The default configuration if none of these options are present is to run
  * the actor on the same executor as its parent.
- *
- * INTERNAL API
  */
-@DoNotInherit
-@InternalApi
-private[akka] sealed trait DispatcherSelector extends Props
+object DispatcherSelector {
+
+  /**
+   * Scala API:
+   * Run the actor on the same executor as its parent.
```

Comment:
That's what the docs of the class said, I'd have to check to say for sure. But I think it makes sense (and has been something users ask every now and then)